### ISSUE TITLE
Excessive commitlog I/O in unit tests

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylla#9058]
Running database_test:

```console
$ du -sm   testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/*
80	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/commitlog.dir
0	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/hints.dir
28	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/ks
1	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/system
0	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/system_auth
0	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/system_distributed
0	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/system_distributed_everywhere
3	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/system_schema
0	testlog/debug/scylla-50d694c0-bef8-4e05-979d-c9efe5c0c10f/view_hints.dir
```

We should consider disabling commitlog for all tests except those that are intended to test commitlog.

**Comment by nyh** [comment]
I agree that for purely functional tests (e.g., whether some CQL feature works), we don't need commit logs, and we don't even need sstable flushes. However, keeping the commit logs (and sstable flushes, compactions, etc.) active means one more (small...) chance of finding bugs in those systems. I think the question is what would be the benefit of disabling commit logs. Did you try to see if the tests run signficantly faster, for example?

I am guessing that tests that involve a lot of schema changes (creating tables, etc.) are slowed down considerably by sstable flushing (for example), because each schema change involves multiple sstable flushes (see #4075, #8309). But we should measure whether this is a tiny slowdown or a huge slowdown. I don't know how much the commitlogs specifically hurt.

**Comment by avikivity** [comment]
I didn't try, but I see that my system is sometimes blocked on I/O rather than on CPU, even in the debug tests.
